tests/reftests/conflict-badversion.test

```diff
@@ -14,31 +14,9 @@ Faking installation of ocaml.4.02.3
-  - ocaml-beta (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → ocaml (< 4.01.0 | >= 4.03.0) → ocaml-variants != 3.08.5~)
-      unmet availability conditions: enable-ocaml-beta-repository
-  - ocaml-beta (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → custom_printf < 111.21.00 → ocaml < 4.02.0 → ocaml-variants != 3.08.5~)
-      unmet availability conditions: enable-ocaml-beta-repository
-  - ocaml-beta (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → core_kernel = 111.21.00 → ocaml < 4.02.1 → ocaml-variants != 3.08.5~)
-      unmet availability conditions: enable-ocaml-beta-repository
-  - ocaml-system (< 3.08.5~ | >= 3.09.1) (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → ocaml (< 4.01.0 | >= 4.03.0))
-      unmet availability conditions, e.g. sys-ocaml-version = "4.10.0"
-  - ocaml-system (< 3.08.5~ | >= 3.09.1) (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → custom_printf < 111.21.00 → ocaml < 4.02.0)
-      unmet availability conditions, e.g. sys-ocaml-version = "4.10.0"
-  - ocaml-system (< 3.08.5~ | >= 3.09.1) (from core (> 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00 & > 112.17.00 & < 112.17.00) → core_kernel = 111.21.00 → ocaml < 4.02.1)
-      unmet availability conditions, e.g. sys-ocaml-version = "4.10.0"
-Your request can't be satisfied:
-  - No available version of core satisfies the constraints
-  - No available version of ocaml satisfies the constraints
-  - No available version of ocaml-base-compiler satisfies the constraints
-  - ocaml-base-compiler (< 3.07+1 | = 3.08.2 | = 3.08.4 | = 3.09.1 | >= 3.10.0 & < 3.10.2 | = 3.11.1 | = 3.12.0 | = 4.02.3) is in conflict with ocaml-variants
-  - ocaml-base-compiler = 4.02.3 is in conflict with ocaml-system
+[ERROR] Package conflict!
+  * No agreement on the version of core:
+    - core != 112.17.00
+    - core != 112.17.00
```

Comment:
what does it mean? :-)

Reply:
The message is off, but there is actually little we can easily do about it :/ ; my opinion is that in this case the error is clear anyway so it's still OK (but yeah, we should at least filter out duplicate lines).

As for the explanation why, this is a different instance of #2229: we are working on a package graph where we can't really distinguish between conjunction and disjunction between edges. In this case, we have `core<112.17.00 & core>112.17.00`, which actually resolves to `false`, but without knowing about the `&` we wrongly reduce it as an `|`, which reduces to `core != 112.17.00`.
I'll check if I can find a way to detect such cases, and skip the "formula reduction" when they happen.

Reply:
ha yes I was talking about the duplication -- but you are right, it's a bit hard to suggest something more actionnable to the user here.